src/nvmet.c

```diff
@@ -595,25 +951,36 @@ int nvmet_unexport_volume(unsigned char *device, unsigned char *host, int port)
 				if (access(path, F_OK) == SUCCESS) {
 					rc = unlink(path);
 					if (rc != SUCCESS) {
-						printf("Error. Unable to remove NQN from ports.\n%s: unlink: %s\n", __func__, strerror(errno));
+						msg = "Error. Unable to remove NQN from ports. %s: unlink: %s";
+						print_error(msg, return_message, __func__, strerror(errno));
+						list = clean_scandir(list, err);
 						return rc;
 					}
 				}
 			}
-			if (list[n] != NULL) free(list[n]);
 		}
-        	printf("Block device %s has been unmapped from all NVMe Target ports.\n", device);
+		list = clean_scandir(list, err);
+		if (return_message == NULL) {
+			msg = "Block device %s has been unmapped from all NVMe Target ports.";
+			print_error(msg, return_message, device);
+		}
 	}
 
 port_check_out:
 
 	/* If a port and / or host are defined, just remove / unmap and exit. */
-	if ((strlen(host) == 0) && (port == INVALID_VALUE)) {
+
+	/* I changed this test from "((strlen(host) == 0) && (port == INVALID_VALUE))"
+	 * because prevented the complete removal of the mapping
+	 */
+
+//	if ((strlen(host) != 0) && (port != INVALID_VALUE)) {
```

Comment:
I need to revisit this and figure out what I did what I did here and the problem I was intending to solve.

Reply:
Ok!